src/extension/background.js

```diff
@@ -78,30 +85,54 @@ const finishVisit = tabId => {
 };
 
 const messageListener = port => message => {
-  const tabId = port.sender.tab.id;
-  let theme;
-  switch (message.type) {
-    case "fetchTheme":
-      log("fetchTheme");
-      fetchTheme().then(({ theme: currentTheme }) =>
-        port.postMessage({ type: "fetchedTheme", theme: currentTheme })
-      );
-      break;
-    case "setTheme":
-      theme = normalizeTheme(message.theme);
-      log("setTheme", theme);
-      storeTheme({ theme });
-      applyTheme({ theme });
-      break;
-    case "setMetrics":
-      siteTabMetrics[tabId] = message.params;
-      log("setMetrics for tab", tabId);
-      break;
-    case "ping":
-      port.postMessage({ type: "pong" });
-      break;
-    default:
-      log("unexpected message", message);
+  const { type } = message;
+  messageHandlers[type in messageHandlers ? type : "default"](message, port);
```

Comment:
Moved away from a switch statement to a function dispatch table, because the local variable situation started getting hairy.